### PR TITLE
[TypeScript][Virtual Assistant] fix: localize qna maker dialog IDs

### LINKED ISSUE #3496: QnAMakerDialog is not created for different languages
#### What project is affected?

Virtual Assistant and Skills

#### What language is this in?

Typescript 

#### What happens?

When a **QnAMakerDialog** is created with a specific language (ex. DE) will be resolving always that language even though you change the language or another user uses a different language (ex EN).

#### What are the steps to reproduce this issue?

**User 1**: Sends utterance to the bot ( DE );
**Bot**: The Bot resolves the utterance as QnA intent, and create the new QnADialog, with the knowledgebaseId as Id of the Dialog:

   ```
 private registerQnADialog(knowledgebaseId: string, cognitiveModels: ICognitiveModelSet, locale: string): void {
        const qnaEndpoint: QnAMakerEndpoint | undefined = cognitiveModels.qnaConfiguration.get(knowledgebaseId);
        if (qnaEndpoint == undefined){
            throw new Error(`Could not find QnA Maker knowledge base configuration with id: ${ knowledgebaseId }.`);
        }

        if (this.dialogs.find(knowledgebaseId) == undefined) {
            const qnaDialog: QnAMakerDialog = new QnAMakerDialog(
                qnaEndpoint.knowledgeBaseId,
                qnaEndpoint.endpointKey,
                // The following line is a workaround until the method getQnAClient of QnAMakerDialog is fixed
                // as per issue https://github.com/microsoft/botbuilder-js/issues/1885
                new URL(qnaEndpoint.host).hostname.split('.')[0],
                this.templateEngine.generateActivityForLocale('UnsupportedMessage', locale) as Activity,
                // Before, instead of 'undefined' a '0.3' value was used in the following line
                undefined,
                this.templateEngine.generateActivityForLocale('QnaMakerAdaptiveLearningCardTitle', locale).text,
                this.templateEngine.generateActivityForLocale('QnaMakerNoMatchText', locale).text
            );

            qnaDialog.id = knowledgebaseId;

            this.addDialog(qnaDialog);
        }
    }
``` 

**Bot**: Sends QnA Answer to User 1 in User 1's locale ( DE )
**User 2**: Sends utterance to bot ( EN );
**Bot**: The Bot resolves the utterance as QnA intent ( same the user 1 sent ), no new QnaMakerDialog is created because one Dialog with the same knowledgebaseId already exists.
**Bot**: The Bot start again the QnA Dialog (added from user 1 iteration) and get the answer from the Knowledge Base (DE) of the first user;
**Bot**: Sends Message to User 2 in User 1's locale.

#### What were you expecting to happen?

Another **QnAMakerDialog** will be created with the right language and right knowledge base ID.

### PR DESCRIPTION
**Pull request by joshgummersall**
Fixes #3496

### CODE REVIEW

File: templates/typescript/samples/sample-assistant/src/dialogs/mainDialog.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ import { BotServices } from '../services/botServices';
 import { IBotSettings } from '../services/botSettings';
 import { StateProperties } from '../models/stateProperties';
 import { OnboardingDialog } from './onboardingDialog';
+import { QnaMakerService } from 'botframework-config';
```
Comment (Batta32) [inline templates/typescript/samples/sample-assistant/src/dialogs/mainDialog.ts]: Is this `import` necessary? It's not used in the `MainDialog` and in the template is not added.

**Review by Batta32 — APPROVED** [review]
LGTM! Just a minor comment.

File: templates/typescript/samples/sample-assistant/src/dialogs/mainDialog.ts (hunk 1 above)
Comment (joshgummersall) [reply templates/typescript/samples/sample-assistant/src/dialogs/mainDialog.ts]: It was a remnant of an earlier change that I backed out. Fixed!

**Review by Batta32 — APPROVED** [review]

**Comment by Batta32** [comment]
Hi @joshgummersall, we were using these changes internally and we noticed that for the [`faq`](https://github.com/microsoft/botframework-solutions/blob/600caea8cd20baa1a0a6bcd9916e647dee9b9be8/templates/typescript/generator-bot-virtualassistant/generators/app/templates/sample-assistant/src/dialogs/mainDialog.ts#L375) dialog IDs the issue still happens.

We fixed the problem replicating the changes for that scenario. Can we apply the same changes for this case?

_Replicated changes for q_faq_
image

**Comment by joshgummersall** [comment]
@Batta32, rather than duplicate the code, I refactored the common elements to a helper method. This should achieve the same thing, though.

**Comment by srinaath** [comment]
@lauren-mills can we get another review on this PR to get it merged?

**Review by lauren-mills — APPROVED** [review]

**Comment by joshgummersall** [comment]
@lauren-mills, is there one more person I can request to review this PR? It looks like I need two approving reviews to merge.

**Pull request merged**